include/envoy/runtime/runtime.h

```diff
@@ -102,6 +102,19 @@ class Snapshot {
   virtual bool featureEnabled(const std::string& key, uint64_t default_value,
                               uint64_t random_value) const PURE;
 
+  /**
+   * Test if a feature is enabled using the built in random generator and total number of buckets
+   * for sampling.
+   * @param key supplies the feature key to lookup.
+   * @param default_value supplies the default value that will be used if either the feature key
+   *        does not exist or it is not an integer.
+   * @param num_buckets control max number of buckets for sampling. Sampled value will be in a range
+   * of [0, num_buckets).
+   * @return true if the feature is enabled.
+   */
+  virtual bool sampleFeatureEnabled(const std::string& key, uint64_t default_value,
```

Comment:
nit: I see why you had to change the name here, but IMO "sample" is a bit strange in the sense that the other featureEnabled() functions also do sampling. A few options here:
1) Get rid of the versions of featureEnabled() that don't require specifying number of buckets. Larger change.
2) Some other name. `featureEnabledEx` ? (Bad, but not sure what would be better).

Reply:
@mattklein123 thanks for reviewing! I've addressed your other comments in f197fe2, and I'm working on this now.

Reply:
Implemented option 1 in f98bcc0.